lib/iris/_lazy_data.py

```diff
@@ -69,21 +69,40 @@ def as_lazy_data(data, chunks=_MAX_CHUNK_SIZE):
     if not is_lazy_data(data):
         if isinstance(data, ma.MaskedArray):
             data = array_masked_to_nans(data)
-            data = data.data
         data = da.from_array(data, chunks=chunks)
     return data
 
 
-def array_masked_to_nans(array, mask=None):
+def array_masked_to_nans(masked_array):
```

Comment:
> the argument name should be changed

Good point! Shows how much change there's been to this function recently...

> Possibly we should change the function name too

To what though? Primarily what it does is put `nan`s where before there were masked values, which is what the function says it does. If we wanted to describe the non-masked array pass through too then we might end up with:
`array_masked_to_nans_unless_the_array_isnt_masked_at_which_point_just_return_it`
... which is a bit long (and very Pratchett-ian! :tophat:)

Do you have a better name suggestion? Otherwise I propose we don't touch it.